core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java

```diff
@@ -249,4 +257,67 @@ long getMaxNumberOfThreads() {
 
     }
 
+    static class MaxMapCountCheck implements Check {
+
+        private final long limit = 1 << 18;
```

Comment:
I dont agree, most will be from file based memory maps, which is a fraction of the amount of files. Like i said, if we do this, we should just go to full mmapdirectory because we lose the benefits of reducing the number of memory maps there: mainly that users don't need to worry about it anymore.